modules/rostests/apitests/ntdll/NtSetInformationProcess.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * PROJECT:         ReactOS API tests
+ * LICENSE:         GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:         Tests for the NtSetInformationProcess API
+ * COPYRIGHT:       Copyright 2020 Bișoc George <fraizeraust99 at gmail dot com>
+ */
+
+#include "precomp.h"
+
+static
+void
+Test_ProcForegroundBackgroundClass(void)
+{
+    NTSTATUS Status;
+    PPROCESS_FOREGROUND_BACKGROUND ProcForeground;
+
+    ProcForeground = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, sizeof(PROCESS_FOREGROUND_BACKGROUND));
+    if (ProcForeground == NULL)
+    {
+        skip("Failed to allocate memory block from heap for PROCESS_FOREGROUND_BACKGROUND! Skipping...\n");
+        return;
+    }
+
+    /* As a test, set the foreground of the retrieved current process to FALSE */
+    ProcForeground->Foreground = FALSE;
+
+    /* Set everything NULL for the caller */
+    Status = NtSetInformationProcess(NULL,
+                                     ProcessForegroundInformation,
+                                     NULL,
+                                     0);
+    ok_hex(Status, STATUS_INFO_LENGTH_MISMATCH);
+
+    /* Give an invalid process handle (but the input buffer and length are correct) */
+    Status = NtSetInformationProcess(NULL,
+                                     ProcessForegroundInformation,
+                                     ProcForeground,
+                                     sizeof(PROCESS_FOREGROUND_BACKGROUND));
+    ok_hex(Status, STATUS_INVALID_HANDLE);
+
+    /* Give a buffer data to the argument input as NULL */
+    Status = NtSetInformationProcess(NtCurrentProcess(),
+                                     ProcessForegroundInformation,
+                                     NULL,
+                                     sizeof(PROCESS_FOREGROUND_BACKGROUND));
+    ok_hex(Status, STATUS_ACCESS_VIOLATION);
+
+    /* Set the information process foreground with an incorrect buffer alignment and zero buffer length */
+    Status = NtSetInformationProcess(NtCurrentProcess(),
+                                     ProcessForegroundInformation,
+                                     (PVOID)1,
+                                     0);
+    ok_hex(Status, STATUS_INFO_LENGTH_MISMATCH);
```

Comment:
In case you didn't get the point of what I'm trying to do, I'm doing some multiple combinations in NtSetInformationProcess() call to see how this function behaves. This is self-explanatory, really.